--- DVDStyler-1.5.1/data/dvdstyler.desktop~	2007-06-13 08:55:19.000000000 +0300
+++ DVDStyler-1.5.1/data/dvdstyler.desktop	2007-07-08 22:43:02.000000000 +0300
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-Encoding=iso-8859-1
 Name=DVD Styler
 Comment=Video DVD Production
-Icon=dvdstyler.png
+Icon=dvdstyler
 Exec=dvdstyler
 Terminal=false
-Categories=Application;AudioVideo
+Categories=AudioVideo;
 StartupNotify=true
+StartupWMClass=Dvdstyler
