--- data/dvdstyler.desktop.orig	2007-07-29 05:00:10.000000000 -0400
+++ data/dvdstyler.desktop	2009-01-24 16:22:27.000000000 -0500
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-Encoding=UTF-8
 Name=DVD Styler
-Comment=Video DVD Production
-Icon=dvdstyler.png
+Comment=Produce your own video DVD
+Icon=dvdstyler
 Exec=dvdstyler
 Terminal=false
-Categories=Application;AudioVideo;
+Categories=AudioVideo;
 StartupNotify=true
+StartupWMClass=Dvdstyler
