--- configure.in.orig	2008-10-15 22:59:08.000000000 -0400
+++ configure.in	2008-10-15 22:59:21.000000000 -0400
@@ -59,7 +59,7 @@
 
 dnl check for wxSVG
 AC_CHECK_LIB(wxsvg, main, [LIBS="$LIBS -lwxsvg"], [AC_MSG_ERROR([*** missing wxSVG library (wxsvg.sf.net) ***])])
-AC_CHECK_HEADER([wxSVG/svg.h], [], [AC_MSG_ERROR([*** missing wxSVG library (wxsvg.sf.net) ***])])
+AC_CHECK_HEADER([wxSVG-freeworld/svg.h], [], [AC_MSG_ERROR([*** missing wxSVG-freeworld library (rpmfusion.org) ***])])
 
 dnl check gnomeui-2.0
 CHECK_WX_BUILT_WITH_GTK2()
--- ./wxVillaLib/ThumbnailFactory.cpp.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./wxVillaLib/ThumbnailFactory.cpp	2008-09-27 14:31:15.000000000 -0400
@@ -27,7 +27,7 @@
 #include <libgnomeui/gnome-thumbnail.h>
 static GnomeThumbnailFactory* thumbnail_factory = NULL;
 #elif defined(WX_SVG)
-#include <wxSVG/mediadec_ffmpeg.h>
+#include <wxSVG-freeworld/mediadec_ffmpeg.h>
 #elif defined(FFMPEG)
 #include "mediadec_ffmpeg.h"
 #endif
--- ./src/dvdstyler.cpp.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/dvdstyler.cpp	2008-09-27 14:31:15.000000000 -0400
@@ -17,8 +17,8 @@
 #include <wxVillaLib/imagjpg.h>
 #include <wxVillaLib/utils.h>
 #include <wxVillaLib/ThumbnailFactory.h>
-#include <wxSVG/SVGCanvas.h>
-#include <wxSVG/mediadec_ffmpeg.h>
+#include <wxSVG-freeworld/SVGCanvas.h>
+#include <wxSVG-freeworld/mediadec_ffmpeg.h>
 
 #ifndef __WXWINCE__
 #include <locale.h>
--- ./src/MenuObject.cpp.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/MenuObject.cpp	2008-09-27 14:31:14.000000000 -0400
@@ -11,8 +11,8 @@
 #include "Menu.h"
 
 #include <wxVillaLib/utils.h>
-#include <wxSVG/svg.h>
-#include <wxSVGXML/svgxmlhelpr.h>
+#include <wxSVG-freeworld/svg.h>
+#include <wxSVGXML-freeworld/svgxmlhelpr.h>
 #include <wx/mstream.h>
 #include <wx/filename.h>
 
--- ./src/Utils.h.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/Utils.h	2008-09-27 14:31:14.000000000 -0400
@@ -11,7 +11,7 @@
 #ifndef UTILS_H
 #define UTILS_H
 
-#include <wxSVGXML/svgxml.h>
+#include <wxSVGXML-freeworld/svgxml.h>
 
 const wxString DATAFORMAT_MENU =  wxT("application/dvdstyler-menu");
 const wxString DATAFORMAT_MENU_OBJECT =  wxT("application/dvdstyler-menu-object");
--- ./src/DVD.h.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/DVD.h	2008-09-27 14:31:14.000000000 -0400
@@ -14,7 +14,7 @@
 #include <wx/wx.h>
 #include <wx/image.h>
 #include <wx/dynarray.h>
-#include <wxSVG/mediadec_ffmpeg.h>
+#include <wxSVG-freeworld/mediadec_ffmpeg.h>
 #include "Menu.h"
 #include "Slideshow.h"
 
--- ./src/MenuEditor.h.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/MenuEditor.h	2008-09-27 14:31:13.000000000 -0400
@@ -11,7 +11,7 @@
 #ifndef MENUEDITOR_H
 #define MENUEDITOR_H
 
-#include <wxSVG/svgctrl.h>
+#include <wxSVG-freeworld/svgctrl.h>
 #include "DVD.h"
 class TitlesetManager;
 
--- ./src/Menu.cpp.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/Menu.cpp	2008-09-27 14:31:13.000000000 -0400
@@ -13,9 +13,9 @@
 #include "Palette3D.h"
 #include <wxVillaLib/SConv.h>
 #include <wxVillaLib/utils.h>
-#include <wxSVG/svg.h>
-#include <wxSVG/SVGCanvas.h>
-#include <wxSVGXML/svgxmlhelpr.h>
+#include <wxSVG-freeworld/svg.h>
+#include <wxSVG-freeworld/SVGCanvas.h>
+#include <wxSVGXML-freeworld/svgxmlhelpr.h>
 #include <wx/mstream.h>
 #include <wx/filename.h>
 
--- ./src/Slideshow.cpp.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/Slideshow.cpp	2008-09-27 14:31:13.000000000 -0400
@@ -9,7 +9,7 @@
 //////////////////////////////////////////////////////////////////////////////
 
 #include "Slideshow.h"
-#include <wxSVGXML/svgxmlhelpr.h>
+#include <wxSVGXML-freeworld/svgxmlhelpr.h>
 
 
 //////////////////////////////////////////////////////////////////////////////
--- ./src/ProgressDlg.cpp.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/ProgressDlg.cpp	2008-09-27 14:31:12.000000000 -0400
@@ -17,8 +17,8 @@
 #include <wx/filename.h>
 #include <wx/file.h>
 #include <wx/regex.h>
-#include <wxSVG/SVGDocument.h>
-#include <wxSVG/mediadec_ffmpeg.h>
+#include <wxSVG-freeworld/SVGDocument.h>
+#include <wxSVG-freeworld/mediadec_ffmpeg.h>
 #include "mediaenc_ffmpeg.h"
 #include "mediatrc_ffmpeg.h"
 
--- ./src/MenuEditor.cpp.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/MenuEditor.cpp	2008-09-27 14:31:12.000000000 -0400
@@ -17,8 +17,8 @@
 #include <wxVillaLib/ImageProc.h>
 #include <wxVillaLib/Thumbnails.h>
 #include <wxVillaLib/utils.h>
-#include <wxSVG/svg.h>
-#include <wxSVGXML/svgxml.h>
+#include <wxSVG-freeworld/svg.h>
+#include <wxSVGXML-freeworld/svgxml.h>
 #include "math.h"
 #include <wx/dnd.h>
 #include <wx/utils.h>
--- ./src/DVD.cpp.orig	2008-09-27 14:29:41.000000000 -0400
+++ ./src/DVD.cpp	2008-09-27 14:31:11.000000000 -0400
@@ -10,10 +10,10 @@
 
 #include "DVD.h"
 #include "Config.h"
-#include <wxSVGXML/svgxmlhelpr.h>
+#include <wxSVGXML-freeworld/svgxmlhelpr.h>
 #include <wxVillaLib/utils.h>
 #include <sys/stat.h>
-#include <wxSVG/mediadec_ffmpeg.h>
+#include <wxSVG-freeworld/mediadec_ffmpeg.h>
 
 #define DATA_FILE(fname) wxFindDataFile(_T("data") + wxString(wxFILE_SEP_PATH) + fname)
 
