diff -up src/mediatrc_ffmpeg.cpp.ff src/mediatrc_ffmpeg.cpp
--- src/mediatrc_ffmpeg.cpp.ff	2008-08-24 22:34:02.000000000 +0200
+++ src/mediatrc_ffmpeg.cpp	2008-10-15 22:25:03.000000000 +0200
@@ -958,8 +958,8 @@ bool wxFfmpegMediaTranscoder::Run(bool& 
 
 			if (!codec->codec_tag) {
 				if ( !os->oformat->codec_tag || av_codec_get_id(
-						os->oformat->codec_tag, icodec->codec_tag) > 0
-						|| av_codec_get_tag(os->oformat->codec_tag,
+						const_cast<const struct AVCodecTag **>(os->oformat->codec_tag), icodec->codec_tag) > 0
+						|| av_codec_get_tag(const_cast<const struct AVCodecTag **>(os->oformat->codec_tag),
 								icodec->codec_id) <= 0)
 					codec->codec_tag = icodec->codec_tag;
 			}
